eva/third_party/huggingface/binder.py

```diff
@@ -12,28 +12,28 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from eva.catalog.catalog_type import NdArrayType
 from eva.catalog.models.udf_catalog import UdfCatalogEntry
-from eva.third_party.huggingface.model import ImageHFModel, TextHFModel
+from eva.third_party.huggingface.create import MODEL_FOR_TASK
 
 
 def assign_hf_udf(udf_obj: UdfCatalogEntry):
     """
     Assigns the correct HF Model to the UDF. The model assigned depends on
-    the type of input the UDF expects. This is done so that we can
+    the task type for the UDF. This is done so that we can
     process the input correctly before passing it to the HF model.
     """
     inputs = udf_obj.args
 
     # NOTE: Currently, we only support models that require a single input.
     assert len(inputs) == 1, "Only single input models are supported."
-    input_type = inputs[0].array_type
 
-    # By default, we assume that the input is an image.
-    model_class = ImageHFModel
+    task = None
+    for metadata in udf_obj.metadata:
+        if metadata.key == "task":
+            task = metadata.value
+            break
 
```

Comment:
Can probably make this a helper function, `get_metadata_entry(<key>)` that can be reused in other parts of the code. I believe we will use this for caching and batching as well.